websites/S/StreamKiste/presence.ts

```diff
@@ -0,0 +1,218 @@
+            {
+              label:
+                strings[
+                  {
+                    movie: "buttonViewMovie",
+                    series: "buttonViewSeries",
+                    other: "buttonViewPage"
+                  }[type]
+                ],
+              url: document.location.href
+            }
+          ];
+          return data;
+        }
+      },
+      {
+        middleware: (ref) => !!ref.location.pathname.match(/^\/(cat)\//i),
+        exec: (context, data, { strings, images }: ExecutionArguments) => {
+          if (!context) return null;
+          data.details = strings.searching;
+          if (
+            document
+              .querySelector(`.category .arg > .type`)
+              ?.textContent.toLowerCase()
+              .indexOf("genre") === 0
+          ) {
+            data.state = `${strings.viewGenre} ${
+              document.querySelector(`.category .arg > .q.arg_category`)
+                ?.textContent
+            }`;
+          } else {
+            delete data.state;
+          }
+          data.smallImageKey = images.SEARCH;
+          return data;
+        }
+      },
+      {
+        middleware: (ref) => !!ref.location.hostname.match(/streamkiste/i),
+        exec: (context, data, { strings, images }: ExecutionArguments) => {
+          if (!context) return null;
+          data.details = strings.browsing;
+          delete data.state;
+          data.smallImageKey = images.BROWSE;
+          return data;
+        }
+      }
+    ],
+    presence = new Presence({
+      clientId: "825531268581818419"
+    });
+
+  (function (app: Presence) {
+    let lastPageIndex: number,
+      currentLang: string,
+      localizedStrings: { [key: string]: string },
+      frameData: VideoContext;
+    const IMAGES = {
+      LOGO: "logox1024",
+      PLAY: "playx1024",
+      PAUSE: "pausex1024",
+      BROWSE: "browsex1024",
+      SEARCH: "searchx1024"
+    };
+    app.on("iFrameData", (data: VideoContext) => {
+      frameData = data;
+    });
+    app.on("UpdateData", async () => {
+      const newLang = await app.getSetting("lang");
+      if (newLang !== currentLang) {
+        currentLang = newLang;
+        localizedStrings = await app.getStrings(
+          {
+            browsing: "presence.activity.browsing",
+            watchingMovie: "general.watchingMovie",
+            watchingSeries: "general.watchingSeries",
+            buttonViewMovie: "general.buttonViewMovie",
+            buttonViewSeries: "general.buttonViewSeries",
+            buttonViewPage: "general.buttonViewPage",
+            viewGenre: "general.viewGenre",
+            playing: "general.playing",
+            searching: "general.search",
+            searchFor: "general.searchFor"
+          },
+          newLang
+        );
+      }
+      const presenceData: PresenceData = {
+          largeImageKey: IMAGES.LOGO,
+          largeImageText: app.metadata.service
+        } as PresenceData,
```

Comment:
: PresenceData 
as PresenceData

Shouldnt be needed double??

Reply:
![image](https://user-images.githubusercontent.com/17952364/112760816-95092d00-8ff8-11eb-9249-9099736b3acb.png)
some properties do not exist, as PresenceData overrides the error and marks the obj valid

Reply:
Cause you cannot set the largeImageKey